src/data/finset.lean

```diff
@@ -677,6 +691,24 @@ end
 @[simp] lemma sdiff_subset (s t : finset α) : s \ t ⊆ s :=
 by simp [subset_iff, mem_sdiff] {contextual := tt}
 
+lemma union_sdiff_distrib (s₁ s₂ t : finset α) : (s₁ ∪ s₂) \ t = s₁ \ t ∪ s₂ \ t :=
+by { simp only [ext, mem_sdiff, mem_union], tauto }
+
+lemma sdiff_union_distrib (s t₁ t₂ : finset α) : s \ (t₁ ∪ t₂) = (s \ t₁) ∩ (s \ t₂) :=
+by { simp only [ext, mem_union, mem_sdiff, mem_inter], tauto }
+
+lemma union_sdiff_self (s t : finset α) : (s ∪ t) \ t = s \ t :=
+by rw [union_sdiff_distrib, sdiff_self, union_empty]
+
+lemma sdiff_singleton_eq_erase (a : α) (s : finset α) : s \ singleton a = erase s a :=
+by { ext, rw [mem_erase, mem_sdiff, mem_singleton], tauto }
+
+lemma sdiff_sdiff_self_left (s t : finset α) : s \ (s \ t) = s ∩ t :=
+by { simp only [ext, mem_sdiff, mem_inter], tauto }
+
+lemma sdiff_partially_injective {s t₁ t₂ : finset α} : s \ t₁ = s \ t₂ → s ∩ t₁ = s ∩ t₂ :=
```

Comment:
I don't like this name because it can't be guessed. What about the longer but clearer `inter_eq_inter_of_sdiff_eq_sdiff`?